src/unittests/f32_interval.spec.ts

```diff
@@ -300,134 +304,990 @@ g.test('contains_interval')
-      { intervals: [[kValue.f32.negative.nearest_min, 0]], expected: [kValue.f32.negative.nearest_min, 0]},
-      { intervals: [[kValue.f32.infinity.negative, 0]], expected: [Number.NEGATIVE_INFINITY, 0]},
+      { intervals: [[0, 10]], expected: [0, 10] },
+      { intervals: [[0, kValue.f32.positive.max]], expected: [0, kValue.f32.positive.max] },
+      { intervals: [[0, kValue.f32.positive.nearest_max]], expected: [0, kValue.f32.positive.nearest_max] },
+      { intervals: [[0, kValue.f32.infinity.positive]], expected: [0, Number.POSITIVE_INFINITY] },
+      { intervals: [[kValue.f32.negative.min, 0]], expected: [kValue.f32.negative.min, 0] },
+      { intervals: [[kValue.f32.negative.nearest_min, 0]], expected: [kValue.f32.negative.nearest_min, 0] },
+      { intervals: [[kValue.f32.infinity.negative, 0]], expected: [Number.NEGATIVE_INFINITY, 0] },
+
+      // Double Intervals
+      { intervals: [[0, 1], [2, 5]], expected: [0, 5] },
+      { intervals: [[2, 5], [0, 1]], expected: [0, 5] },
+      { intervals: [[0, 2], [1, 5]], expected: [0, 5] },
+      { intervals: [[0, 5], [1, 2]], expected: [0, 5] },
+      { intervals: [[kValue.f32.infinity.negative, 0], [0, kValue.f32.infinity.positive]], expected: kAny },
+
+      // Multiple Intervals
+      { intervals: [[0, 1], [2, 3], [4, 5]], expected: [0, 5] },
+      { intervals: [[0, 1], [4, 5], [2, 3]], expected: [0, 5] },
+      { intervals: [[0, 1], [0, 1], [0, 1]], expected: [0, 1] },
+
+      // Point Intervals
+      { intervals: [1], expected: 1 },
+      { intervals: [1, 2], expected: [1, 2] },
+      { intervals: [-10, 2], expected: [-10, 2] },
+    ]
+  )
+  .fn(t => {
+    const intervals = t.params.intervals.map(toF32Interval);
+    const expected = toF32Interval(t.params.expected);
+
+    const got = F32Interval.span(...intervals);
+    t.expect(
+      objectEquals(got, expected),
+      `span({${intervals}}) returned ${got}. Expected ${expected}`
+    );
+  });
+
+interface IsF32VectorCase {
+  input: (number | IntervalBounds | F32Interval)[];
+  expected: boolean;
+}
+
+g.test('isF32Vector')
+  .paramsSubcasesOnly<IsF32VectorCase>([
+    // numbers
+    { input: [1, 2], expected: false },
+    { input: [1, 2, 3], expected: false },
+    { input: [1, 2, 3, 4], expected: false },
+
+    // IntervalBounds
+    { input: [[1], [2]], expected: false },
+    { input: [[1], [2], [3]], expected: false },
+    { input: [[1], [2], [3], [4]], expected: false },
+    {
+      input: [
+        [1, 2],
+        [2, 3],
+      ],
+      expected: false,
+    },
+    {
+      input: [
+        [1, 2],
+        [2, 3],
+        [3, 4],
+      ],
+      expected: false,
+    },
+    {
+      input: [
+        [1, 2],
+        [2, 3],
+        [3, 4],
+        [4, 5],
+      ],
+      expected: false,
+    },
+
+    // F32Interval, valid dimensions
+    { input: [toF32Interval([1]), toF32Interval([2])], expected: true },
```

Comment:
Would be nice if some of these provided something other than a single `number` to `toF32Interval`